orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.java

```diff
@@ -0,0 +1,649 @@
+import static redis.clients.jedis.BinaryClient.LIST_POSITION.BEFORE;
+
+@Component
+@Slf4j
+public class JedisExecutionRepository extends AbstractRedisExecutionRepository {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  @Autowired
+  public JedisExecutionRepository(
+    Registry registry,
+    @Qualifier("redisClientDelegate") RedisClientDelegate redisClientDelegate,
+    @Qualifier("previousRedisClientDelegate") Optional<RedisClientDelegate> previousRedisClientDelegate,
+    @Qualifier("queryAllScheduler") Scheduler queryAllScheduler,
+    @Qualifier("queryByAppScheduler") Scheduler queryByAppScheduler,
+    @Value("${chunkSize.executionRepository:75}") Integer threadPoolChunkSize
+  ) {
+    super(registry, redisClientDelegate, previousRedisClientDelegate, queryAllScheduler, queryByAppScheduler, threadPoolChunkSize);
+  }
+
+  public JedisExecutionRepository(
+    Registry registry,
+    RedisClientDelegate redisClientDelegate,
+    Optional<RedisClientDelegate> previousRedisClientDelegate,
+    Integer threadPoolSize,
+    Integer threadPoolChunkSize
+  ) {
+    super(
+      registry,
+      redisClientDelegate,
+      previousRedisClientDelegate,
+      Schedulers.from(Executors.newFixedThreadPool(10)),
+      Schedulers.from(Executors.newFixedThreadPool(threadPoolSize)),
+      threadPoolChunkSize
+    );
+  }
+
+  @Override
+  public void store(@Nonnull Execution execution) {
+    getRedisDelegateForId(execution.getId()).withTransaction(tx -> {
+      storeExecutionInternal(tx, execution);
+      if (execution.getType() == PIPELINE) {
+        tx.zadd(executionsByPipelineKey(
+          execution.getPipelineConfigId()),
+          execution.getBuildTime() != null ? execution.getBuildTime() : currentTimeMillis(),
+          execution.getId());
+      }
+      tx.exec();
+    });
+  }
+
+  @Override
+  public void storeStage(@Nonnull Stage stage) {
+    ExecutionType type = stage.getExecution().getType();
+    String key = format("%s:%s", type.toString(), stage.getExecution().getId());
+    getRedisDelegateForId(key).withTransaction(tx -> {
+      storeStageInternal(tx, type, stage);
+      tx.exec();
+    });
+  }
+
+  @Override
+  public void removeStage(@Nonnull Execution execution, @Nonnull String stageId) {
+    String key = format("%s:%s", execution.getType().toString(), execution.getId());
+    String indexKey = format("%s:stageIndex", key);
+    getRedisDelegateForId(key).withCommandsClient(c -> {
+      List<String> stageIds = new ArrayList<>(Arrays.asList(c.hget(key, "stageIndex").split(",")));
+      stageIds.remove(stageId);
+      boolean hasDiscreteIndex = c.exists(indexKey);
+      List<String> keys = c.hkeys(key).stream()
+        .filter(k -> k.startsWith("stage." + stageId))
+        .collect(Collectors.toList());
+
+      getRedisDelegateForId(key).withTransaction(tx -> {
+        tx.hset(key, "stageIndex", stageIds.stream().map(Object::toString).collect(Collectors.joining(",")));
```

Comment:
Why do we need `.map(Object::toString)`? They _are_ strings.

Reply:
Derp, replaced with `stageIds.stream().collect(Collectors.joining(",")))`